### ISSUE TITLE
implementation bug in has_failure()

### ISSUE THREAD

**Issue by BurningEnlightenment** [ned14/outcome#162]
`has_failure()` implementation from [basic_result_final.hpp#L71 ](https://github.com/ned14/outcome/blob/develop/include/outcome/detail/basic_result_final.hpp#L71) doesn't match its description `returns True if has error or exception.`:
```c++
return (this->_state._status & detail::status_have_error) != 0 && (this->_state._status & detail::status_have_exception) != 0;
                                                               ^
```

It should probably be:
```c++
return (this->_state._status & (detail::status_have_error | detail::status_have_exception)) != 0;
```

**Comment by ned14** [comment]
Fixed! Thanks for reporting this bug.

**Comment by BurningEnlightenment** [comment]
@ned14 the critical change hasn't been included in 94064e5 🙈

**Comment by ned14** [comment]
Thank you **so** much for spotting this. It's fixed now. I have *no* idea how this fix did not make it into that commit. Very worrying.
